### PR TITLE
motoc: init at 0.2.0

### PR DESCRIPTION
**Pull request by Pandapip1**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by myclevorname** [comment]
I built it for the target aarch64-linux on my x86_64-linux machine, and it ran.

File: pkgs/by-name/mo/motoc/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "motoc";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "galister";
+    repo = "motoc";
+    rev = "refs/tags/v${version}";
```
Comment (myclevorname) [inline pkgs/by-name/mo/motoc/package.nix:16]:
```suggestion
    rev = "v${version}";
```

You don't need `refs/tags/`.

**Review by myclevorname — CHANGES_REQUESTED** [review]

File: pkgs/by-name/mo/motoc/package.nix (hunk 1 above)
Comment (Pandapip1) [reply pkgs/by-name/mo/motoc/package.nix:16]: I prefer to be explicit, to the point of nitting on other PRs to add the `refs/tags/` prefix.

**Review by myclevorname — APPROVED** [review]
LGTM

**Review by Scrumplex — APPROVED** [review]

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 340172` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>motoc</li>
  </ul>
</details>

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/mo/motoc/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "MOnado Tracking Origin Calibration program";
+    homepage = "https://github.com/galister/motoc";
+    changelog = "https://github.com/galister/motoc/releases";
+    license = lib.licenses.gpl3Plus;
+    maintainers = with lib.maintainers; [ pandapip1 ];
+    mainProgram = "motoc";
+    platforms = lib.platforms.all;
```
Comment (Aleksanaa) [inline pkgs/by-name/mo/motoc/package.nix:41]:
```suggestion
```

Rust has its available platforms in buildRustPackage by default

Comment (Pandapip1) [reply pkgs/by-name/mo/motoc/package.nix:41]: Good to know, thanks. I'll address that in a future PR, I'd like to get this merged first.

**Review by myclevorname — APPROVED** [review]

**Pull request merged**